core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileSourceConfig.java

```diff
@@ -155,15 +149,28 @@ protected void validate() {
       (format.equalsIgnoreCase("avro")) || format.equalsIgnoreCase("parquet"))) {
       throw new IllegalArgumentException("Format can only be 'text', 'avro' or 'parquet'");
     }
-    if (!format.equalsIgnoreCase("text")) {
-      try {
-        new Schema.Parser().parse(inputSchema);
-      } catch (Exception e){
-        throw new IllegalArgumentException("Unable to parse schema with error: " + e.getMessage(), e);
+    Schema parsedSchema;
+    try {
+      parsedSchema = Schema.parseJson(schema);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Unable to parse schema with error: " + e.getMessage(), e);
+    }
+
+    if (format.equalsIgnoreCase("text")) {
+      if (schema != null && !schema.isEmpty()) {
+        if (!(schema.contains("offset") && schema.contains("body"))) {
```

Comment:
you need to check the types.  check that offset is a long or nullable long and body is a string or nullable string.

```
Schema.Field offsetField = schema.getField("offset");
if (offsetField == null) {...}
Schema offsetSchema = offsetField.getSchema();
Schema.Type offsetType = offsetSchema.isNullable() ? offsetSchema.getNonNullable().getType() : offsetSchema.getType();
if (offsetType != Schema.Type.LONG) { ... }
```